doc/Guide to Contributing to Sia.md

```diff
@@ -0,0 +1,175 @@
+To build Sia on your machine, enter the following on the command line:
+```bash
+# Download Sia and its dependencies
+# Binaries will be installed in $GOROOT/bin
+$ go get -u github.com/NebulousLabs/Sia/...
+
+# Switch to directory containing Sia source code
+$ cd $GOROOT/src/github.com/NebulousLabs/Sia
+
+# You have three Sia builds to choose from.
+# To build the standard release binary:
+$ make release-std
+# Or to build the release binary with race detection and an array debugging asserts:
+$ make release
+# Or to build the developer binary (with a different genesis block, faster block times, and other changes):
+$ make
+```
+
+<a name="contribute"/>
+## Contribute to the codebase
+<a name="setup-git"/>
+### Set up git
+Install git on your machine according to [these instructions][install-git] in the Pro Git book.
+
+You will first need to set up global settings using the command line.
+```bash
+$ git config --global user.name "Your Name"
+$ git config --global user.email you@somedomain.com
+
+# Tell git to remember your login information for a certain amount of time.
+# Default time is 15 minutes:
+$ git config --global credential.helper cache
+# Or you can choose a different amount of time:
+$ git config --global credential.helper "cache --timeout=[seconds]"
+
+```
+<a name="fork"/>
+### Fork the Sia repository
+While logged into your Github account, navigate to the [Sia repository][sia] and click the 'Fork' button in the upper righthand corner.  Your account now has a 'forked' copy of the original repo at `https://github.com/[your github username]/Sia`.
+
+Clone this forked repo into a local directory:
+```bash
+$ cd $GOPATH/src/github.com/[your github username]
+$ git clone https://github.com/[your github username]/Sia
+```
+Add the original repository as a Git remote called 'upstream'.
+```bash
+$ git remote add upstream https://github.com/NebulousLabs/Sia.git
+```
+Now you can keep your local copy up to date with the original repo by running
+```bash
+$ git pull upstream master
+```
+<a name="write">
+### Write some code
+Right now your git local repository only has one branch (called 'master' by default).  Add a new branch and make your changes in that one:
+
+```bash
+# Create and checkout (switch to) a new branch called my_new_feature
+$ git checkout -b my_new_feature
+```
+Only implement one logical change per branch. If you're working on several things at once, make multiple branches.  To learn more about branching, see [Using the Fork-and-Branch Git Workflow][branch] and [Pro Git - Branches in a Nutshell][nutshell].
+
+Be sure to follow the conventions detailed in [docs/Developers.md][developers.md].  We will reject pull requests that do not satisfy these best practices.
+
+Once you've finished making changes, stage and commit your changes then update your Github repository:
+```bash
+# Make sure you're on the right branch
+$ git checkout my_new_feature
+# Use the gofmt tool to format your code automatically
+$ gofmt .
```

Comment:
`go fmt` is preferred, but even better would be `make test`, since this runs `go fmt` *and* ensures that you haven't broken anything.